src/app.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import Form from './form.js'
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      cards: []
+    }
+    this.addCard = this.addCard.bind(this)
+  }
+  addCard(card) {
+    const addCard = this.state.cards.map(card => Object.assign({}, card))
```

Comment:
No need to copy all of the other cards when adding one. You just need to copy the `Array`.